### PR TITLE
Author preview

### PR DESCRIPTION
**Pull request by eessex**
- Adds author_ids field to article schema
- Adds ability to query authors by name
- Adds UI to add an author to an article

A few ideas to chew on for future PRs - 
- updating model to acknowledge this field on editorial articles
- updating distribution logic to use this field over contributing author

### CODE REVIEW

File: api/apps/articles/test/model/index/persistence.test.coffee [hunk 1]
```diff
@@ -590,6 +590,16 @@ describe 'Article Persistence', ->
         article.author.name.should.equal 'Jon Snow'
         done()
 
+    it 'saves the author_ids field', (done) ->
+      Article.save {
+        author_ids: [ '5086df098523e60002000018', '5086df098523e60002000015' ]
+      }, 'foo', {}, (err, article) ->
+        return done err if err
+        console.log(article.author_ids)
```
Comment (kanaabe) [inline api/apps/articles/test/model/index/persistence.test.coffee]: stray log

File: api/apps/authors/model.coffee [hunk 2]
```diff
@@ -40,8 +40,13 @@ Joi = require '../../lib/joi'
...
 
 @mongoFetch = (input, callback) ->
-  cursor = db.authors
-    .find({})
+  query = omit input, 'q', 'limit', 'offset', 'count', 'strict'
+  if input.strict
+    query.name = { $eq: input.q } if input.q and input.q.length
+  else
+    query.name = { $regex: ///#{input.q}///i } if input.q and input.q.length
+    cursor = db.authors
```
Comment (kanaabe) [inline api/apps/authors/model.coffee]: Hmm, I'm confused about this indentation?

File: api/apps/articles/test/model/index/persistence.test.coffee (hunk 1 above)
Comment (eessex) [reply api/apps/articles/test/model/index/persistence.test.coffee]: o thanks

**Comment by eessex** [comment]
The client has a few methods to grab and print the article author -- was thinking specifically of this place, and to look into if there are others -- https://github.com/artsy/positron/blob/master/client/models/article.coffee#L47

**Comment by kanaabe** [comment]
Yes, that makes sense. Would you mind adding that point (and the distribution one) to GH somewhere? We'll definitely want to tackle these before we launch.

**Pull request merged**